src/Swagger/ApiDocumentationBuilder.php

```diff
@@ -0,0 +1,293 @@
+
+            $class = [
+                'name' => $shortName,
+                'externalDocs' => [
+                    'url' => $prefixedShortName,
+                ],
+            ];
+
+            if ($description = $resourceMetadata->getDescription()) {
+                $class = [
+                    'name' => $shortName,
+                    'description' => $description,
+                    'externalDocs' => [
+                        'url' => $prefixedShortName,
+                    ],
+                ];
+            }
+
+            $attributes = $resourceMetadata->getAttributes();
+            $context = [];
+
+            if (isset($attributes['normalization_context']['groups'])) {
+                $context['serializer_groups'] = $attributes['normalization_context']['groups'];
+            }
+
+            if (isset($attributes['denormalization_context']['groups'])) {
+                $context['serializer_groups'] = isset($context['serializer_groups']) ? array_merge($context['serializer_groups'], $attributes['denormalization_context']['groups']) : $context['serializer_groups'];
+            }
+
+            foreach ($this->propertyNameCollectionFactory->create($resourceClass, $context) as $propertyName) {
+                $propertyMetadata = $this->propertyMetadataFactory->create($resourceClass, $propertyName);
+
+                if ($propertyMetadata->isIdentifier() && !$propertyMetadata->isWritable()) {
+                    continue;
+                }
+
+                $property[$propertyName] = [
+                    'type' => $this->getRange($propertyMetadata),
+                ];
+
+                $required = [];
+
+                if ($propertyMetadata->isRequired()) {
+                    $required = array_merge($required, [$propertyName]);
+                }
+
+                if (0 !== count($required)) {
+                    $properties[$shortName]['required'] = $required;
+                }
+
+                $properties[$shortName]['type'] = 'object';
+                $properties[$shortName]['properties'] = $property;
+            }
+
+            if ($operations = $resourceMetadata->getItemOperations()) {
+                foreach ($operations as $operationName => $itemOperation) {
+                    $itemOperations = array_merge($itemOperations, $this->getSwaggerOperation($resourceClass, $resourceMetadata, $operationName, $itemOperation, $prefixedShortName, false));
+                }
+            }
+            $itemOperationsDocs[$this->iriConverter->getIriFromResourceClass($resourceClass)] = $itemOperations;
+            $classes[] = $class;
+        }
+
+        $doc['swagger'] = '2.0';
+        if ('' !== $this->title) {
+            $doc['info']['title'] = $this->title;
+        }
+
+        if ('' !== $this->description) {
+            $doc['info']['description'] = $this->description;
+        }
+        $doc['host'] = $_SERVER['HTTP_HOST'];
```

Comment:
Super-globals should never be accessed directly. This data would better be injected in the constructor.